internal/provider/resource_docker_service.go

```diff
@@ -23,7 +23,7 @@ func resourceDockerService() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"auth": {
-				Type:     schema.TypeMap,
+				Type:     schema.TypeList,
```

Comment:
State Migration should be implemented.

Reply:
https://www.terraform.io/docs/extend/resources/state-migration.html#terraform-v0-12-sdk-state-migrations

Reply:
Thanks for pointing out :) So for all `TypeMap` to `TypeList` we need those migrators right?

Reply:
I think so.

Reply:
- tested locally with unit tests and
- as described in the `contributing` section (I wonder how we can automate those tests)
```hcl
terraform {
  required_providers {
    docker = {
      source  = "kreuzwerker/docker"
      #version = "2.8.0"
      version = "9.9.9"
    }
  }
}

provider "docker" {
}

resource "docker_image" "foo" {
  name         = "nginx:latest"
  keep_locally = true
}

resource "docker_service" "foo" {
  name = "foo"
  task_spec {
    container_spec {
      image             = docker_image.foo.latest
      stop_grace_period = "10s"
    }
    # v2.8.0
    #restart_policy = { 
    # v9.9.9
    restart_policy {
      condition    = "on-failure"
      delay        = "3s"
      max_attempts = 4
      window       = "10s"
    }

  }

  endpoint_spec {
    ports {
      target_port    = "80"
      published_port = "8080"
    }
  }
}
```

Reply:
Great work!
Would you split files per schema version like `docker_container` resource?

* resource_docker_service.go
* resource_docker_service_v1.go
* resource_docker_service_v0.go

I did the same in thing https://github.com/hashicorp/terraform-provider-docker/pull/272 .